src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder_ImplementingTypes.cs

```diff
@@ -0,0 +1,78 @@
+            Solution solution,
+            IImmutableSet<Project> projects,
+            bool transitive,
+            CancellationToken cancellationToken)
+        {
+            return FindTypesFromCacheOrComputeAsync(
+                type, solution, projects,
+                transitive ? s_typeToTransitivelyImplementingTypesMap : s_typeToImmediatelyImplementingTypesMap,
+                c => FindWithoutCachingImplementingTypesAsync(type, solution, projects, transitive, c),
+                cancellationToken);
+        }
+
+        private static async Task<ImmutableArray<INamedTypeSymbol>> FindWithoutCachingImplementingTypesAsync(
+            INamedTypeSymbol type,
+            Solution solution,
+            IImmutableSet<Project> projects,
+            bool transitive,
+            CancellationToken cancellationToken)
+        {
+            // Only an interface can be implemented.
+            if (type?.TypeKind == TypeKind.Interface)
+            {
+                // Note: it is intentional that we do TypeHasBaseTypeInSet even though we're searching for interfaces
+                // (which are never in the BaseType chain of a type).  The reason for this is the following case:
+                //
+                //  interface IGoo { }
+                //
+                //  class Base : IGoo { }
+                //
+                //  class Derived : Base { }
+                //
+                // In this case, IGoo has two implementations (Base and Derived).  When searching we'll first find the
+                // 'Base' match and will add to the set.  Then, we'll look for types that have 'Base' in their
+                // inheritance chain, and we need to match that by looking in the .BaseType inheritance chain when
+                // looking at 'Derived'.
+                static bool TypeMatches(INamedTypeSymbol type, SymbolSet set)
+                    => TypeHasBaseTypeInSet(type, set) || TypeHasInterfaceInSet(type, set);
+
+                // As long as we keep hitting derived interfaces or implementing non-sealed classes we need to keep
+                // looking (as those types themselves may have more derived classes that would b implementations of this
+                // interface).  If we hit structs/sealed-classes though we can stop as they can't have any more types
+                // that inherit from them.
+                var allTypes = await DescendInheritanceTreeAsync(type, solution, projects,
+                    typeMatches: TypeMatches,
+                    shouldContinueSearching: s_isInterfaceOrNonSealedClass,
+                    transitive: transitive,
+                    cancellationToken: cancellationToken).ConfigureAwait(false);
+
+                // Only classes/struct implement interface types.  Derived interfaces can be found with
+                // FindDerivedInterfacesAsync.
+                return allTypes.WhereAsArray(t => t.TypeKind == TypeKind.Class || t.TypeKind == TypeKind.Struct);
```

Comment:
Very interesting case!  It will come down to: does the compiler model this in the .Interfaces list of the enum.  if so, then i should find these. if not, then i think it's fine not to.